src/linux.rs

```diff
@@ -300,8 +300,17 @@ impl<'a> I2CTransfer<'a> for LinuxI2CDevice {
 
     /// Issue the provided sequence of I2C transactions
     fn transfer(&mut self, messages: &'a mut [Self::Message]) -> Result<u32, LinuxI2CError> {
+        let msg_type = |flag: u16| flag & I2CMessageFlags::READ.bits();
+        let mut prev_msg_type = messages.first().map(|m| msg_type(m.flags));
```

Comment:
This should start as `None` correct? That way the first message will always have its START flag set. Then following messages of the same type will have the NO_START bit set.

Reply:
100% correct. Idk what I was thinking when I wrote that. I also forgot that my test application didn't automatically pull the new version so I was testing against the old implementation.